server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -3082,7 +3082,14 @@ public void startRecovery(
                 executeRecovery("from store", recoveryState, recoveryListener, this::recoverFromStore);
                 break;
             case REMOTE_STORE:
-                executeRecovery("from remote store", recoveryState, recoveryListener, this::restoreFromRemoteStore);
+                final Repository remoteTranslogRepo;
```

Comment:
we could set this as null and then get rid of else block?

Reply:
is it possible to push down some of this logic to `restoreFromRemoteStore` method?

Reply:
`remoteTranslogRepo` needs to be final as it is provided as argument to a lambda, so we can't initialize it twice.

`restoreFromRemoteStore` does not have reference to `repositoriesService` to fetch the repository.

Reply:
Can we please use `Optional` instead?